django/template/loaders/cached.py

```diff
@@ -18,9 +18,7 @@ def __init__(self, engine, loaders):
         super().__init__(engine)
 
     def get_dirs(self):
-        for loader in self.loaders:
-            if hasattr(loader, "get_dirs"):
-                yield from loader.get_dirs()
+        return [dir for loader in self.loaders for dir in loader.get_dirs() if loader]
```

Comment:
To the best of my _recollection_, `get_dirs` isn't **required** on a loader, hence the existing code doing the `hasattr` check, which was added in 29845ecf6999e9e31b56a82229d885f8dfb99e04 as part of [ticket #25791](https://code.djangoproject.com/ticket/25791) which is _more recent_ than the PR you've rebased -- as such, I'd assume this bit can/should be reverted/unnecessary?

Reply:
Ah, yes. I was making the change based on 57864dc125a8ae1c624da5cbb6b1cc3d1171c4af